astropy/coordinates/builtin_frames/utils.py

```diff
@@ -211,7 +222,26 @@ def aticq(ri, di, astrom):
     d = np.zeros_like(pnat)
     for j in range(5):
         before = norm(pnat-d)
-        after = erfa.ld(1.0, before, before, astrom['eh'], astrom['em'], 5e-8)
+        if ignore_distance:
+            # No distance to object, assume a long way away
+            q = before
+        else:
+            # Find BCRS direction of Sun to object.
+            # astrom['eh'] and astrom['em'] contain Sun to observer unit vector,
+            # and distance, respectively.
+            eh = astrom['em'] * CartesianRepresentation(astrom['eh'], unit=u.au, xyz_axis=-1, copy=False)
```

Comment:
I love representations and quantities, but I think here I would not use them, since everything is basically in the right format already. So, just
```
q = (astrom['em'][..., np.newaxis] * astrom['eh']
     + srepr.distance[..., np.newaxis].to_value(u.au) * before)
sundist = np.linalg.norm(q, axis=-1, keepdims=True)
q /= sundist
q[sundist < 1.e-10] = before
```
(and, as noted above, maybe get the distance out on top, perhaps even in AU?)

Reply:
Thanks for the suggestion - and the solution! It really does take me three goes at least to get shapes right 🥇 

I stuck with ```np.where``` for the last line, since the direct mask struggled with trailing dimensions.